core/src/test/resources/sql/misc.iq

```diff
@@ -3182,4 +3182,41 @@ from (values 0) as t(x);
 EnumerableValues(tuples=[[{ true, false, true, false, true, false, true, false }]])
 !plan
 
+# [CALCITE-6560] Add supportsNegativeScale in RelDataTypeSystem
+
+!set outputformat csv
+
+# throw an exception when CAST target decimal type has illegal precision and scale parameter
+
+# precision is negative number
+values cast(15.3 as decimal(-1,4));
+Error while executing SQL "values cast(15.3 as decimal(-1,4))": parse failed: Encountered "-" at line 1, column 29.
+!error
+
+# precision is 0
+values cast(15.3 as decimal(0,4));
+DECIMAL precision 0 must be between 1 and max precision 19
+!error
+
+# scale is negative number
+values cast(15.3 as decimal(1,-1));
+Error while executing SQL "values cast(15.3 as decimal(1,-1))": parse failed: Encountered "-" at line 1, column 31.
```

Comment:
this suggests that the parser has to be changed to allow negative values for scale

Reply:
Yes. I have tried to do that. But this PR will become complicated. In Calcite， it has the same specific processing about Scala. So I left it to another PR when set supportsNegativeScale is true.

Reply:
I don't think this is a big change. It's probably enough to change this line in the Parser.jj file:
```
scale = UnsignedIntLiteral()
```
to
```
scale = IntLiteral()
```

Reply:
Yes, But when we execute SQL :
```
select cast(13.4 as decima(3,-1)) 
```
will be converted to
```
select cast(13.4 as decima(3,0)) 
```
So, we can't throw exceptions about the don't support negative scale.

Reply:
it means that there is at least one more bug somewhere - this shouldn't happen

Reply:
If we support a negative scale, then we can start by Parser.jj.

Reply:
I will do it in another new issue.

Reply:
I hope it's not a lot of code, if you add it to this issue you can exercise these tests properly

Reply:
it's a one line change to Parser.jj, and you have to find out where the scale is lost. that may be everything you need to change.

Reply:
Ok. I will check it.